open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/common/idempotency/IdempotencyValidationUtils.java

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.wso2.openbanking.accelerator.consent.extensions.common.idempotency;
+
+import com.wso2.openbanking.accelerator.common.config.OpenBankingConfigParser;
+import com.wso2.openbanking.accelerator.common.exception.ConsentManagementException;
+import com.wso2.openbanking.accelerator.consent.extensions.internal.ConsentExtensionsDataHolder;
+import com.wso2.openbanking.accelerator.consent.extensions.manage.model.ConsentManageData;
+import com.wso2.openbanking.accelerator.consent.mgt.service.ConsentCoreService;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Map;
+
+/**
+ * Class to hold idempotency validation utils.
+ */
+public class IdempotencyValidationUtils {
+
+    private static final Log log = LogFactory.getLog(IdempotencyValidationUtils.class);
+    private static final Map<String, Object> configs = OpenBankingConfigParser.getInstance().getConfiguration();
+    private static final ConsentCoreService consentCoreService = ConsentExtensionsDataHolder.getInstance()
+            .getConsentCoreService();
+
+    /**
+     * Method to check whether the required parameters for idempotency validation are present.
+     *
+     * @param consentManageData Consent Manage Data
+     * @return Whether the required parameters are present
+     */
+    static boolean isMandatoryParamsPresent(ConsentManageData consentManageData, String idempotencyKeyName) {
+        String idempotencyKeyValue = consentManageData.getHeaders().get(idempotencyKeyName);
+        return idempotencyKeyValue != null && consentManageData.getPayload() != null &&
+                consentManageData.getClientId() != null;
```

Comment:
Isn't it better to do these checks individually and log the respective failure reason?

Reply:
Fixed in https://github.com/wso2/financial-open-banking/pull/51/commits/e1205691228fc4b76853a0eab7dd99dac2516661